libnavigation-ui/src/main/java/com/mapbox/navigation/ui/speed/SpeedLimitView.java

```diff
@@ -0,0 +1,214 @@
+package com.mapbox.navigation.ui.speed;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Color;
+import android.graphics.drawable.GradientDrawable;
+import android.graphics.drawable.LayerDrawable;
+import android.text.SpannableStringBuilder;
+import android.text.style.RelativeSizeSpan;
+import android.text.style.StyleSpan;
+import android.util.AttributeSet;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.Nullable;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.lifecycle.LifecycleObserver;
+
+import com.mapbox.navigation.base.speed.model.SpeedLimit;
+import com.mapbox.navigation.base.speed.model.SpeedLimitSign;
+import com.mapbox.navigation.base.speed.model.SpeedLimitUnit;
+import com.mapbox.navigation.ui.R;
+
+import static android.graphics.Typeface.BOLD;
+import static android.text.Spanned.SPAN_EXCLUSIVE_EXCLUSIVE;
+
+public class SpeedLimitView extends ConstraintLayout implements LifecycleObserver {
+
+  private static final double KILO_MILES_FACTOR = 0.621371;
+  private static final float RADIUS = 10f;
+  private static final int VIENNA_MUTCD_BORDER_INSET = 4;
+  private static final int MUTCD_INNER_BACKGROUND_INSET = 7;
+  private static final int VIENNA_INNER_BACKGROUND_INSET = 16;
+  private static final int VIENNA_MUTCD_OUTER_BACKGROUND_INSET = 0;
+  private static final int MUTCD_BORDER_COLOR = Color.BLACK;
+  private static final int VIENNA_BORDER_COLOR = Color.RED;
+  private TextView speedLimitText;
+  private ImageView speedLimitView;
+  private ConstraintLayout speedView;
+  private int speedLimitBackgroundColor;
+
+  public SpeedLimitView(Context context) {
+    this(context, null);
+  }
+
+  public SpeedLimitView(Context context, @Nullable AttributeSet attrs) {
+    this(context, attrs, -1);
+  }
+
+  public SpeedLimitView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+    super(context, attrs, defStyleAttr);
+    inflate(getContext(), R.layout.layout_speed_limit, this);
+    initAttributes(attrs);
+  }
+
+  /**
+   * Invoke to set the speed limit
+   * @param speedLimit object storing speed limit data
+   */
+  public void setSpeedLimit(@Nullable SpeedLimit speedLimit) {
+    if (speedLimit != null) {
```

Comment:
Let's ask from the non-null value from the get-go if we're not handling null anyway.